src/controller/lib/handle.js

```diff
@@ -12,23 +14,23 @@ module.exports = function (config, baseUrl) {
     })
   );
 
-  return function (request, response) {
-    const server = 'express'; // detect if hapi or express here
-    const process = requestHandler[method].bind(requestHandler);
-    const format = payloadHandler[method].bind(payloadHandler, config);
-    const respond = (responder ? responder : send[server]).bind(null, response);
-    const errors = requestHandler.validate(request);
+  const validate = requestHandler.validate.bind(requestHandler);
+  const process = requestHandler[method].bind(requestHandler);
+  const formatPayload = payloadHandler[method].bind(payloadHandler, config);
 
-    if (errors) {
-      respond(payloadHandler.error(errors));
-    } else {
-      process(request)
-        .then(format)
-        .then(respond)
-        .catch(function (err) {
-          //throw err;
-          return respond(payloadHandler.error(err));
-        });
-    }
+  var buildPayload = function (request) {
+    var errors = validate(request);
+    return bPromise[errors ? 'reject' : 'resolve'](errors)
```

Comment:
I'd like to be able to unit test this directly. Can we move it into the `lib` folder and add the arguments needed so we can partially apply it here?